FlashLFQ/FlashLFQResults.cs

```diff
@@ -299,6 +300,181 @@ public void CalculateProteinResultsTop3(bool useSharedPeptides)
+            foreach (ProteinGroup proteinGroup in ProteinGroups.Values)
+            {
+                if (proteinGroupToPeptides.TryGetValue(proteinGroup, out var peptidesForThisProtein))
+                {
+                    // calculate reference protein intensity
+                    double referenceProteinIntensity = 0;
+                    int topNPeaks = 3;
+
+                    foreach (var group in filesGroupedByCondition)
+                    {
+                        List<double> highestIntensities = new List<double>();
+
+                        foreach (var peptide in peptidesForThisProtein)
+                        {
+                            double max = 0;
+
+                            foreach (var sample in group)
+                            {
+                                double peptideIntensity = peptide.GetIntensity(sample);
+
+                                if (!double.IsNaN(peptideIntensity))
+                                {
+                                    max = Math.Max(max, peptideIntensity);
+                                }
+                            }
+
+                            highestIntensities.Add(max);
+                        }
+
+                        referenceProteinIntensity = highestIntensities.OrderByDescending(p => p).Take(topNPeaks).Sum();
+
+                        if (referenceProteinIntensity > 0)
+                        {
+                            break;
+                        }
+                    }
+
+                    // set up peptide intensity table
+                    int numSamples = 0;
+                    foreach (var condition in SpectraFiles.GroupBy(p => p.SampleGroup))
+                    {
+                        int conditionSamples = condition.Select(p => p.Sample).Distinct().Count();
+                        numSamples += conditionSamples;
+                    }
+
+                    double[,] peptideIntensityMatrix = new double[peptidesForThisProtein.Count, numSamples];
+
+                    // populate matrix w/ peptide intensities
+                    int sampleN = 0;
+                    foreach (var group in SpectraFiles.GroupBy(p => p.SampleGroup).OrderBy(p => p.Key))
+                    {
+                        foreach (var sample in group.GroupBy(p => p.Sample).OrderBy(p => p.Key))
+                        {
+                            foreach (Peptide peptide in peptidesForThisProtein)
+                            {
+                                double sampleIntensity = 0;
+                                double highestFractionIntensity = 0;
+
+                                foreach (var fraction in sample.GroupBy(p => p.Fraction))
+                                {
+                                    double fractionIntensity = 0;
+                                    int replicatesWithValidValues = 0;
+
+                                    foreach (SpectraFileInfo replicate in fraction.OrderBy(p => p.Replicate))
```

Comment:
i'd like comments for this secition also. i'm not understanding how fraction intensites are being used.

Reply:
added a comment to line 407